pkg/cache/tas_flavor_snapshot.go

```diff
@@ -253,18 +238,18 @@ func (s *TASFlavorSnapshot) findLevelWithFitDomains(levelIdx int, required bool,
 	levelDomains := utilmaps.Values(domains)
 	sortedDomain := s.sortedDomains(levelDomains)
 	topDomain := sortedDomain[0]
-	if s.state[topDomain.id] < count {
+	if topDomain.state < count {
```

Comment:
imo, the function reads better now thanks to the state field vs going via the state map